backend/tracim_backend/applications/upload_permissions/email_manager.py

```diff
@@ -0,0 +1,242 @@
+            translator=translator,
+        )
+        send_email_through(
+            config=self.config, sendmail_callable=email_sender.send_mail, message=message
+        )
+        emails_receivers_list = [
+            upload_permission.email for upload_permission in upload_permission_receivers
+        ]
+        logger.info(
+            self,
+            'Generating upload permission mail from user "{}" to "{}"'.format(
+                emitter.user_id, "".join(emails_receivers_list)
+            ),
+        )
+        for upload_permission in upload_permission_receivers:
+            message = self._notify_receiver(
+                emitter=emitter,
+                workspace=workspace,
+                upload_permission=upload_permission,
+                upload_permission_password_enabled=upload_permission_password_enabled,
+                translator=translator,
+            )
+            send_email_through(
+                config=self.config, sendmail_callable=email_sender.send_mail, message=message
+            )
+
+    def _notify_emitter(
+        self,
+        emitter: User,
+        workspace: Workspace,
+        upload_permission_receivers: typing.List[UploadPermissionInContext],
+        upload_permission_password: str,
+        translator: Translator,
+    ) -> Message:
+        logger.info(
+            self,
+            'preparing email to user "{}" about upload_permission on workspace "{}" info created'.format(
+                emitter.user_id, workspace.workspace_id
+            ),
+        )
+        message = MIMEMultipart("alternative")
+        translated_subject = translator.get_translation(
+            self.config.EMAIL__NOTIFICATION__UPLOAD_PERMISSION_TO_EMITTER__SUBJECT
+        )
+        message["Subject"] = translated_subject.format(
+            website_title=self.config.WEBSITE__TITLE,
+            nb_receivers=len(upload_permission_receivers),
+            workspace_name=workspace.label,
+        )
+        message["From"] = self._get_sender()
+        to_addr = formataddr((emitter.display_name, emitter.email))
+        message["To"] = to_addr
+        html_template_file_path = (
+            self.config.EMAIL__NOTIFICATION__UPLOAD_PERMISSION_TO_EMITTER__TEMPLATE__HTML
+        )
+        context = {
+            "emitter": emitter,
+            "workspace": workspace,
+            "upload_permission_receivers": upload_permission_receivers,
+            "upload_permission_password": upload_permission_password,
+        }
+        body_html = self._render_template(
+            mako_template_filepath=html_template_file_path, context=context, translator=translator
+        )
+
+        part2 = MIMEText(body_html, "html", "utf-8")
+        # Attach parts into message container.
+        # According to RFC 2046, the last part of a multipart message,
+        # in this case the HTML message, is best and preferred.
+        message.attach(part2)
+        return message
+
+    def _notify_receiver(
```

Comment:
same remark here

Reply:
issue about this there #2206
Does concern also share app, so not fixed here.